patches/@endo+patterns+1.1.0.patch

```diff
@@ -0,0 +1,546 @@
+diff --git a/node_modules/@endo/patterns/index.d.ts b/node_modules/@endo/patterns/index.d.ts
+index e4eafce..b44325a 100644
+--- a/node_modules/@endo/patterns/index.d.ts
++++ b/node_modules/@endo/patterns/index.d.ts
+@@ -7,5 +7,6 @@ export { isKey, assertKey, assertScalarKey, isCopySet, assertCopySet, makeCopySe
+ export { bagCompare, setCompare, compareKeys, keyLT, keyLTE, keyEQ, keyGTE, keyGT } from "./src/keys/compareKeys.js";
+ export { elementsIsSuperset, elementsIsDisjoint, elementsCompare, elementsUnion, elementsDisjointUnion, elementsIntersection, elementsDisjointSubtract, setIsSuperset, setIsDisjoint, setUnion, setDisjointUnion, setIntersection, setDisjointSubtract } from "./src/keys/merge-set-operators.js";
+ export { bagIsSuperbag, bagUnion, bagIntersection, bagDisjointSubtract } from "./src/keys/merge-bag-operators.js";
+-export { M, getRankCover, isPattern, assertPattern, matches, mustMatch, isAwaitArgGuard, assertAwaitArgGuard, getAwaitArgGuardPayload, isRawGuard, assertRawGuard, assertMethodGuard, getMethodGuardPayload, getInterfaceMethodKeys, assertInterfaceGuard, getInterfaceGuardPayload, kindOf } from "./src/patterns/patternMatchers.js";
++export { M, getRankCover, isPattern, assertPattern, matches, mustMatch, isAwaitArgGuard, assertAwaitArgGuard, isRawGuard, assertRawGuard, assertMethodGuard, assertInterfaceGuard, kindOf } from "./src/patterns/patternMatchers.js";
++export { getAwaitArgGuardPayload, getMethodGuardPayload, getInterfaceGuardPayload, getInterfaceMethodKeys } from "./src/patterns/getGuardPayloads.js";
+ //# sourceMappingURL=index.d.ts.map
+\ No newline at end of file
+diff --git a/node_modules/@endo/patterns/index.js b/node_modules/@endo/patterns/index.js
+index 9dd42f7..b32e6ea 100644
+--- a/node_modules/@endo/patterns/index.js
++++ b/node_modules/@endo/patterns/index.js
+@@ -60,17 +60,20 @@ export {
+   mustMatch,
+   isAwaitArgGuard,
+   assertAwaitArgGuard,
+-  getAwaitArgGuardPayload,
+   isRawGuard,
+   assertRawGuard,
+   assertMethodGuard,
+-  getMethodGuardPayload,
+-  getInterfaceMethodKeys,
+   assertInterfaceGuard,
+-  getInterfaceGuardPayload,
+   kindOf,
+ } from './src/patterns/patternMatchers.js';
+ 
++export {
++  getAwaitArgGuardPayload,
++  getMethodGuardPayload,
++  getInterfaceGuardPayload,
++  getInterfaceMethodKeys,
++} from './src/patterns/getGuardPayloads.js';
```

Comment:
I'm not confident about how to review a patch in detail, but this part of it seems to correspond to
https://github.com/endojs/endo/pull/2038/files#diff-785fa700ea5ccbedf6df4f2d5f2abc2b68e4e7d87dd2c5638b264c49f1f90509

Reply:
How do we usually track the work to get rid of such a patch when it's no longer needed?

Reply:
This is called out in the [maintainers steps](https://github.com/Agoric/agoric-sdk/blob/d4925683c81b6324fcc3c26ec7a6805ed193a9e0/MAINTAINERS.md?plain=1#L270-L274).

I also plan on staging a commit removing the patch in the [`integration-endo-master` branch, used for nightly integration with endo](https://github.com/Agoric/agoric-sdk/blob/d4925683c81b6324fcc3c26ec7a6805ed193a9e0/.github/actions/restore-node/action.yml#L87-L98).

Reply:
removal staged here: https://github.com/Agoric/agoric-sdk/commit/0104d4d6a1e2973e336967db11a51993805f059f